dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -530,6 +530,7 @@ private boolean createWorkflowsAndVersionsFromDockstoreYml(List<? extends Workfl
 
                         Workflow workflow = createOrGetWorkflow(workflowType, repository, user, workflowName, subclass, gitHubSourceCodeRepo);
                         WorkflowVersion version = addDockstoreYmlVersionToWorkflow(repository, gitReference, dockstoreYml, gitHubSourceCodeRepo, workflow, defaultVersion, yamlAuthors);
+                        workflow.syncMetadataWithDefault();
```

Comment:
> It seems like we should only be syncing if the default version changed or was updated. I suppose it's harmless to do it otherwise, and it will update ones we missed, but still...

I didn't check if the default version changed or was updated because it didn't look like we did any checking for other places where this function is called

Reply:
> I didn't check if the default version changed or was updated because it didn't look like we did any checking for other places where this function is called

I feel like this use case is a little different. At a glance, it looks like the other cases it does the sync are when:
1. It's refreshing the entire workflow/tool
2.  Explicitly setting the default version
3. Saving a hosted entry's version; but saving it, makes it the default, so this is a variant of 2.

In this code, we were notified by GitHub of an update to a specific branch/tag, and we should know if the corresponding Dockstore version being updated is the default or not.

But it won't hurt to do this and the object is loaded into memory, so it shouldn't have a performance impact.